tests/test_serializer.c

```diff
@@ -301,6 +301,7 @@ void test_mqtt_device_id_deserialize(void) {
   const char* json = "{\"device_id\":\"" DEVICE_ID "\", \"trytes\":[\"" TRYTES_2673_1 "\",\"" TRYTES_2673_2 "\"]}";
   const int id_len = 32;
   char device_id[id_len + 1];
+  memset(device_id, 0, id_len + 1);
```

Comment:
How about initializing at line 303 with `char device_id[id_len + 1] = {0}`

Reply:
Doing so may lead to error since `device_id` is a variable-sized object, or shall it we change the declaration to `char device_id[ID_LEN + 1];` ?

Reply:
Sure